src/context-menu-manager.coffee

```diff
@@ -150,6 +147,19 @@ class ContextMenuManager
 
     template
 
+  # Returns an object compatible with `::add()` or `null`.
+  cloneItemForEvent: (item, event) ->
+    return null if item.devMode and not @devMode
+    item = Object.create(item)
+    if typeof item.shouldDisplay is 'function'
+      return null unless item.shouldDisplay(event)
+    item.created?(event)
+    if Array.isArray(item.submenu)
+      item.submenu = item.submenu
+        .map((item) => @cloneItemForEvent(item, event))
```

Comment:
These `item` variables shadow the outer `item`, which made this a bit tough to read the first time.